<launch>

<param name="/use_sim_time" value="true"/>

<node pkg="rev_icp_mapper" type="rev_match_cloud_server" name="matcher_server" output="screen">
  <param name="config" value="/home/mqdao/ros/src/rev_icp_mapper/launch/icp_dynamic_outdoor.yaml" />
</node>

<node pkg="rviz" type="rviz" name="rviz" args="-d $(find rev_icp_mapper)/rviz/kitti_icp_odom.rviz" />

<node pkg="rev_icp_mapper" type="icp_odom_node" name="icp_odom_node" output="screen">
  <remap from="cloud_in" to="/kitti/velo/pointcloud" />
  
  <param name="max_cloud" value="154" />
  <param name="startup_drop" value="1" />
  
  <param name="maxOverlapToMerge" value="0.98" />
  
  <!--cloud preprocess param all are in /velo_link-->
  <param name="pointcloud/box_min_x" value="-1.4" />
  <param name="pointcloud/box_max_x" value="1.4" />
  <param name="pointcloud/box_min_y" value="-2.0" />
  <param name="pointcloud/box_max_y" value="2.0" />
  <param name="pointcloud/num_neighbors" value="25" /> <!--ok: 50-->
  <param name="pointcloud/std_dev_mul" value="0.5" /> <!--ok: 1.0-->
  <param name="pointcloud/leaf_size" value="0.1" />
  <param name="pointcloud/min_x" value="0.5" />
  <param name="pointcloud/max_x" value="40.0" />
  <param name="pointcloud/min_y" value="-15.0" />
  <param name="pointcloud/max_y" value="15.0" />
  <param name="pointcloud/min_z" value="-1.25" />  <!--in frame /velo_link-->
  <param name="pointcloud/max_z" value="1.5" />
</node>

<node pkg="rosbag" type="play" name="player" args="-r 0.1 --clock /home/mqdao/Downloads/KITTI/kitti_2011_09_26_drive_0005_synced.bag" />

</launch>